Marlin/src/lcd/e3v2/proui/dwin.cpp

```diff
@@ -68,10 +69,6 @@
   #include "../../../feature/host_actions.h"
 #endif
 
-#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT)
-  #define HAS_ONESTEP_LEVELING 1
-#endif
```

Comment:
We should keep this define. If you've modified the LCD code to call `G29 P1\nG29 P3` for UBL then UBL can be considered "one step" but the ABL options with `PROBE_MANUALLY` are definitely not one-step leveling procedures. `PROBE_MANUALLY` requires user interaction with the encoder to measure each point and proceed to the next point on click.

Reply:
I do not see anywhere in ProUI that `HAS_ONESTEP_LEVELING` is being utilized. its currently defined like this in two seperate files. even the current repo at mriscoc's ProUI doesnt have that and it has no use for `PROBE_MANUALLY`. when UBL or ABL is used, its automatically understood you have that option...to use it as a "one step" or not. just like Mesh Bed Leveling.  
which is also called Manual Mesh Bed Leveling right? since its manually probing the bed without an actual probe.

what is confusing is the `#define PROBE_MANUALLY` option in configuration.h:
>The "Manual Probe" provides a means to do "Auto" Bed Leveling without a probe.  


it sounds like if you have `MESH_BED_LEVELING`, you would want to also enable this. yet in **inc/Conditionals_LCD.h**:
```y
#if !HAS_LEVELING || ANY(MESH_BED_LEVELING, AUTO_BED_LEVELING_UBL)
  #undef PROBE_MANUALLY
```

but besides that, a user having ABL or UBL should always have the option to "Probe manually" without a probe regardless.

that is how the menu is currently, except for maybe JyersUI. that is a bit confusing to have UBL, but also being able to not having any bed probe defined?

in summary, I think this should be left specifically for `MESH_BED_LEVELING`; if you dont have a probe, that should be your only option for mesh leveling.  
and in regards to Auto bed leveling w/probe, it should always give you that "probe manually" option anyway in the menu/gcode. in otherwords, if it `HAS_LEVELING` the "probing manually" is always an option. 

I'm just saying this is how it is set up with ProUI right now
- Probe?
UBL, or ABL
- No Probe?
Mesh Bed Leveling, or None

I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

It seems that things aren't what they used to be, or meant to be anymore. or maybe I have it all wrong. like was `AUTO_BED_LEVELING_UBL` supposed to be used without a bed probe like it can be in JyersUI? if so, why not just use `MESH_BED_LEVELING` instead?

on a side note, even Mriscoc is phasing out using Bilinear auto bed leveling in favor of UBL. I know their code is different, but on the surface its the same thing. *you probe, and you get a mesh.*  
plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

Reply:
> plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

Currently the paradigm is different, due to the addition of AML 2.0 support, now all mesh parameters are saved in the UBL slot, so you can have slots with different mesh sizes and mesh densities. You can know how it is working now and the new way of use UBL in the Professional firmware here: https://github.com/mriscoc/Ender3V2S1/wiki/Mesh-Leveling-Menu#ubl

Reply:
I picked this line at random to comment on the define, but the general idea is to keep using this define in code that currently uses it, wherever that is.

Reply:
> I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

Yes, that option is specifically for a machine that has no probe. You can also use `PROBE_MANUALLY` with any ABL option for a machine that has no probe. In both cases, you send a `G29` command to get each point, and keep doing this in a loop until the procedure is completed.

See `menu_bed_leveling.cpp` within the `ANY(PROBE_MANUALLY, MESH_BED_LEVELING)` block to see how this is implemented from the UI.

Reply:
> due to the addition of AML 2.0 support

Good that you have a wiki to cover the topic. My aim with ABL Bilinear was to make a leveling system that can be done all in one go based on a mesh that only covers the area of the bed that the probe can reach. It actually is "adaptive" in that you can set the boundaries of the mesh that you want to probe, but it always probes the same number of points.

We have a PR pending to make all leveling systems runtime-configurable in both the boundaries and the number of points in X and Y, but of course it's slow going because every LCD implementation wants to do things its own way. I cannot stress enough how important it is to work with the main Marlin project to standardize these things in the core and not to keep "going it alone." So anything you can do towards that effort would be much appreciated. I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

Reply:
> I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

this quote 😂  
oh good, I'm not alone